File: 485. Max Consecutive Ones/MaxConsecutiveOnes.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:

        maxConsecutiveOnes = 0
        consecutiveOnes = 0
        
        for num in nums:
            
            if num == 1:
                consecutiveOnes += 1
                maxConsecutiveOnes = max(maxConsecutiveOnes, consecutiveOnes)
            else:
                consecutiveOnes = 0

        return maxConsecutiveOnes

File: 485. Max Consecutive Ones/MaxConsecutiveOnes.cs
public class Solution {
    public int FindMaxConsecutiveOnes(int[] nums) {
        
        int maxConsecutiveOnes = 0;
        int consecutiveOnes = 0;
        
        foreach (int num in nums) {
            
            if (num == 1) {
                consecutiveOnes += 1;
                maxConsecutiveOnes = Math.Max(maxConsecutiveOnes, consecutiveOnes);
            }
            else {
                consecutiveOnes = 0;
            }          
        }
        return maxConsecutiveOnes;
    }
}
